parser/src/command.rs

```diff
@@ -58,43 +58,98 @@ impl<'a> Input<'a> {
         }
     }
 
-    pub fn parse_command(&mut self) -> Command<'a> {
-        let start = match find_commmand_start(&self.all[self.parsed..], self.bot) {
+    fn maybe_parse_review(&mut self) -> Option<(Tokenizer<'a>, assign::AssignCommand)> {
+        // Both `r?` and `R?`.
+        let start = match self.all[self.parsed..]
+            .find("r?")
+            .or_else(|| self.all[self.parsed..].find("R?"))
+        {
             Some(pos) => pos,
-            None => return Command::None,
+            None => return None,
         };
         self.parsed += start;
         let mut tok = Tokenizer::new(&self.all[self.parsed..]);
-        assert_eq!(
-            tok.next_token().unwrap(),
-            Some(Token::Word(&format!("@{}", self.bot)))
-        );
-        log::info!("identified potential command");
+        match tok.next_token() {
+            Ok(Some(Token::Word(w))) => {
+                if w != "r" && w != "R" {
+                    // If we're intersecting with something else just exit
+                    log::trace!("received odd review start token: {:?}", w);
+                    return None;
+                }
+            }
+            other => {
+                log::trace!("received odd review start token: {:?}", other);
+                return None;
+            }
+        }
+        match tok.next_token() {
+            Ok(Some(Token::Question)) => {}
+            other => {
+                log::trace!("received odd review start token: {:?}", other);
+                return None;
+            }
+        }
+        log::info!("identified potential review request");
+        match tok.next_token() {
+            Ok(Some(Token::Word(w))) => {
+                let mentions = crate::mentions::get_mentions(w);
+                if let [a] = &mentions[..] {
+                    return Some((
+                        tok,
+                        assign::AssignCommand::User {
+                            username: (*a).to_owned(),
+                        },
+                    ));
+                } else {
+                    log::trace!("{:?} had non-one mention: {:?}", w, mentions);
```

Comment:
I think it might be useful to allow more than a person to be assigned: occasionally I write `r? @foo @bar` and manually assign `@bar` once highfive assigns `@foo`. Highfive was written before GitHub allowed multiple assignees, so I guess that's the reason why highfive didn't support that.

Reply:
I largely agree, though I think it is pretty useful to have just one person assigned at a time -- then there's less "well they'll do it" :)

But I don't want to try to make things better before we make things work; migrating things to triagebot is the first priority, and then once that's done we can enhance.